modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -73,6 +73,9 @@
     /** Annotation classes mapped to validator objects. */
     private Map<Class<? extends Annotation>, Set<Validator<?, ?>>> validators = new HashMap<>();
 
+    /** The flag is true when at least one value was read from a storage, otherwise false. */
+    private volatile boolean bootstrapped;
```

Comment:
Why do you think that this is a good idea to put "bootstrapped" flag into configuration changer? Shouldn't it be a property of the Vault?

Reply:
I changed this fix and now the flag is used internal to determine would default applied or not.

Reply:
Code is rolled back, I resolve this conversation